ES6 classes, arrow functions, string templates

IonicaBizau opened the pull request:
- :art: Replaced the old class-like definition with ES6 `class` declaration.
- Used arrow functions making `self` variables obsolete.
- :fire: Removed the `version` field from the `module.exports` (will add that back if there is a good reason to keep it there). We can get the version from `package.json`.
- :memo: JSDoc comments and [`Blah`](https://github.com/IonicaBizau/blah) template to generate the `README.md` file.
- Generated the license via Blah––you will not need to take care of the license year anymore. Well, you will just run `blah -l license` once per year. :smile: 

To generate the `README.md` file we `blah -r` (or `blah --readme`). To know more about how to use `blah`, check out [this post I wrote](http://ionicabizau.net/blog/25-increasing-productivity-using-blah-2-0-0). :memo: 

Fixes #6. :tada: 

Ohh, and happy new year!!! :fireworks:

IonicaBizau committed:
Use the ES6 classes and arrow functions

IonicaBizau committed:
Cleanup

IonicaBizau committed:
JSDoc comments

IonicaBizau committed:
Ignore my vim temp files

IonicaBizau committed:
Blah template and documentation

IonicaBizau committed:
:memo:

IonicaBizau committed:
:art:

IonicaBizau committed:
Updated the node engine version

IonicaBizau committed:
Generated the license

IonicaBizau committed:
Rebuild docs

IonicaBizau commented on benchtable.js:
```diff
@@ -1,218 +1,235 @@
+     */
+    constructor (name, options) {
+
+        super(name, options);
```
This currently doesn't work. But it will work after https://github.com/bestiejs/benchmark.js/pull/112 gets merged. :grin:

izuzak replied:
I think I see a new version of benchmark.js published on npm: https://www.npmjs.com/package/benchmark

@IonicaBizau do you want to bump the dependency version in this PR now? Once that's done, this is good to go, right? I haven't looked at it closely yet, will do that later today or tomorrow. Thanks again! :zap:

IonicaBizau replied:
> I think I see a new version of benchmark.js published on npm

@izuzak Great! I checked that when I woke up (6 hours ago) and saw that is not published. :smile:

Done in e6f85f86de7e56a1d6f01359dbf0d5da3586a86c and fixed the small issues after migration in the latest commits (replaced `pluck` with `map` in the example files and rebuilt the docs, use [`color-it`](https://github.com/ionicabizau/node-color-it) for colors).

Of course, check it out on your side and give me feedback. :grin:

IonicaBizau committed:
Use super to call the overridden functions

IonicaBizau committed:
Beautify the code :art:

IonicaBizau committed:
:arrow_up: benchmark@2.0.0

IonicaBizau committed:
Use map instead of pluck

IonicaBizau committed:
:art: Use color-it to color the things

IonicaBizau committed:
Updated the screenshot and rebuild docs

izuzak commented on README.md:
````diff
@@ -1,178 +1,229 @@
-Benchtable.prototype.addFunction(name, func);
-```
+### `run(name, fun, options)`
+addFunction
````
I think the above two lines should be merged into:

```
### `addFunction(name, fun, options)`
```

Right?

IonicaBizau replied:
The jsdoc comment was wrong. Addressed in d0b1daa5dd57e2cdc8ad70abefeb894b472a7e33 and rebuilt README. :rocket:

izuzak commented on .blah/README.ejs:
```diff
@@ -0,0 +1,116 @@
+Want to contribute to this project? See information [here](CONTRIBUTING.md).
+
+## Credits
+Benchtable is developed by [Ivan Zuzak](http://ivanzuzak.info) \<izuzak@gmail.com\>;.
```
I think there's an extra `;` at the end of the line.

IonicaBizau replied:
Fixed in dd5640c7d1f65aa54a5da9657fb706a6c3253b64. Btw, this looks weird in Atom! :boom: 

![image](https://cloud.githubusercontent.com/assets/2864371/12097709/cf1a8b16-b325-11e5-860e-a8215bee28f3.png)

Before reporting this, can you double check if this is still an issue? I probably have an old version of Atom.

IonicaBizau replied:
There is already an issue regarding that: https://github.com/atom/markdown-preview/issues/320

izuzak commented on README.md:
````diff
@@ -1,178 +1,229 @@
 
-```javascript
+```js
 // import module
````
I think this can be removed. Notice that the first line below is the same comment: https://github.com/izuzak/benchtable/pull/8/files#diff-04c6e90faac2675aa89e2176d2eec7d8R164

IonicaBizau replied:
:+1: 83d282d491bc886487ecb376f36b519aa9c408f7

izuzak commented on README.md:
```diff
@@ -1,178 +1,229 @@
-// run async
-.run({ 'async': false });
+// Import module
+var BenchTable = require('benchtable'); //require('benchtable');
```
Just wondering -- why did you add the comment here? `//require('benchtable');`

IonicaBizau replied:
I didn't. `blah` added it. :smile: I removed it completely from example files in 83d282d491bc886487ecb376f36b519aa9c408f7.

izuzak commented on README.md:
````diff
@@ -1,178 +1,229 @@
+
+```js
+// Import module
+var BenchTable = require('benchtable'); //require('benchtable');
````
What's the purpose of this comment? `//require('benchtable');`

IonicaBizau replied:
@izuzak It was there before (in the examples), and when running `blah -r` we [replace `../` with `benchtable`](https://github.com/izuzak/benchtable/commit/83d282d491bc886487ecb376f36b519aa9c408f7#diff-799839136677d9151042f5761071508cR17). :grin:

izuzak commented:
Other that those few minor comments -- looks good! Great work, @IonicaBizau -- thanks :boom:

IonicaBizau committed:
Fixed jsdoc comment

IonicaBizau committed:
Fix typo (remove semicolon)

IonicaBizau committed:
Improve the examples

Use simple quotes everywhere.
Improve the README.md output. Remove the duplicated code.
Fix the comments

IonicaBizau committed:
Now using color-it instead of colors

IonicaBizau committed:
:memo:

IonicaBizau commented:
@izuzak Done! :joy: Probably you already noticed. I replaced `"foo".red` with [`colorIt("foo").red().toString()`](https://github.com/IonicaBizau/node-color-it/) because after updating the `cli-table` version we don't have these methods in the string prototype (which is good!), and because, why not–flat colors in terminal? :smile:

izuzak commented on README.md:
```diff
@@ -1,178 +1,203 @@
+var suite = new BenchTable();
+
+suite
+  benchtable/ Add functions for benchmarking
```
I think there's something wrong with this line -- there's a single `/` and I think there should be two because it should be a comment, right? And it should be in a different place?

IonicaBizau replied:
@izuzak Oh, great catch! The regex was replacing the things like crazy. Addressed it in https://github.com/IonicaBizauKitchen/benchtable/commit/1a00dea7dd630054ed9dae4ca3a820d2528597da#diff-799839136677d9151042f5761071508cR17 :tada:

IonicaBizau committed:
Fix the regex to replace the `../` with the module name

izuzak commented:
:sparkles: Thanks again for this, @IonicaBizau -- you're a :star2:!

IonicaBizau commented:
@izuzak You're welcome! :blush: Glad to contribute. :tada:

I usually `npm i benchtable` when need to compare two functions and see which one is faster, since I stay all day in the terminal. :smile:
